plugins/chat/assets/javascripts/discourse/modifiers/chat/resizable-node.js

```diff
@@ -0,0 +1,104 @@
+import Modifier from "ember-modifier";
+import { registerDestructor } from "@ember/destroyable";
+import { bind } from "discourse-common/utils/decorators";
+import { throttle } from "@ember/runloop";
+
+const MINIMUM_SIZE = 20;
+
+export default class ResizableNode extends Modifier {
+  element = null;
+  resizerSelector = null;
+  didResizeContainer = null;
+
+  _originalWidth = 0;
+  _originalHeight = 0;
+  _originalX = 0;
+  _originalY = 0;
+  _originalMouseX = 0;
+  _originalMouseY = 0;
+
+  constructor(owner, args) {
+    super(owner, args);
+    registerDestructor(this, (instance) => instance.cleanup());
+  }
+
+  modify(element, [resizerSelector, didResizeContainer]) {
+    this.resizerSelector = resizerSelector;
+    this.element = element;
+    this.didResizeContainer = didResizeContainer;
+
+    this.element
+      .querySelector(this.resizerSelector)
+      ?.addEventListener("mousedown", this._startResize);
+  }
+
+  cleanup() {
+    this.element
+      .querySelector(this.resizerSelector)
+      ?.removeEventListener("mousedown", this._startResize);
+  }
+
+  @bind
+  _startResize(event) {
+    event.preventDefault();
+
+    this._originalWidth = parseFloat(
+      getComputedStyle(this.element, null)
+        .getPropertyValue("width")
+        .replace("px", "")
+    );
+    this._originalHeight = parseFloat(
+      getComputedStyle(this.element, null)
+        .getPropertyValue("height")
+        .replace("px", "")
+    );
+    this._originalX = this.element.getBoundingClientRect().left;
+    this._originalY = this.element.getBoundingClientRect().top;
+    this._originalMouseX = event.pageX;
+    this._originalMouseY = event.pageY;
+
+    window.addEventListener("mousemove", this._resize);
+    window.addEventListener("mouseup", this._stopResize);
+  }
+
+  @bind
+  _resize(event) {
+    throttle(this, this._resizeThrottled, event, 24);
+  }
+
+  @bind
+  _resizeThrottled(event) {
```

Comment:
I think it would be great for other people working with this code in the future to add a brief explanation of this calculation. It can be daunting to work on it for the first time.

Reply:
Yes that's fair, will try to document a little bit good call :+1: